pkg/chains/storage/storage_test.go

```diff
@@ -1,68 +0,0 @@
-/*
-Copyright 2020 The Tekton Authors
-Licensed under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License.
-You may obtain a copy of the License at
-    http://www.apache.org/licenses/LICENSE-2.0
-Unless required by applicable law or agreed to in writing, software
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
-*/
-
-package storage
-
-import (
-	"reflect"
-	"testing"
-
-	"github.com/tektoncd/chains/pkg/config"
-	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
-	fakepipelineclient "github.com/tektoncd/pipeline/pkg/client/injection/client/fake"
-	"k8s.io/apimachinery/pkg/util/sets"
-	fakekubeclient "knative.dev/pkg/client/injection/kube/client/fake"
-	logtesting "knative.dev/pkg/logging/testing"
-	rtesting "knative.dev/pkg/reconciler/testing"
-)
-
-func TestInitializeBackends(t *testing.T) {
-
-	tests := []struct {
-		name string
-		cfg  config.Config
-		want []string
-	}{{
-		name: "none",
-		want: []string{},
-	}, {
-		name: "tekton",
-		want: []string{"tekton"},
-		cfg:  config.Config{Artifacts: config.ArtifactConfigs{TaskRuns: config.Artifact{StorageBackend: sets.NewString("tekton")}}},
-	}, {
-		name: "multi",
-		want: []string{"tekton"},
-		cfg:  config.Config{Artifacts: config.ArtifactConfigs{TaskRuns: config.Artifact{StorageBackend: sets.NewString("tekton", "mock")}}},
-	}}
-	logger := logtesting.TestLogger(t)
-	ctx, _ := rtesting.SetupFakeContext(t)
-	ps := fakepipelineclient.Get(ctx)
-	kc := fakekubeclient.Get(ctx)
-	tr := &v1beta1.TaskRun{}
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			got, err := InitializeBackends(ctx, ps, kc, logger, tr, tt.cfg)
-			if err != nil {
-				t.Errorf("InitializeBackends() error = %v", err)
-				return
-			}
-			gotTypes := []string{}
-			for _, g := range got {
-				gotTypes = append(gotTypes, g.Type())
-			}
-			if !reflect.DeepEqual(gotTypes, tt.want) {
-				t.Errorf("InitializeBackends() = %v, want %v", gotTypes, tt.want)
-			}
-		})
-	}
-}
```

Comment:
Are we certain that this is all captured by the existing tests? Do we need to do anything extra for the "multi" test case?

Reply:
Thanks for pointing out this. In previous implementation, I had a `factory.go` file and removed this test storage file. 
Just recovered this file, and added more individual tests and more items to `multi` test :D